src/VisualStudio/Core/Def/Implementation/FindReferences/VisualStudioDefinitionsAndReferencesFactory.cs

```diff
@@ -27,27 +30,31 @@ internal class VisualStudioDefinitionsAndReferencesFactory
         : DefaultDefinitionsAndReferencesFactory
     {
         private readonly IServiceProvider _serviceProvider;
+        private readonly IThreadingContext _threadingContext;
 
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public VisualStudioDefinitionsAndReferencesFactory(SVsServiceProvider serviceProvider)
-            => _serviceProvider = serviceProvider;
+        public VisualStudioDefinitionsAndReferencesFactory(
+            SVsServiceProvider serviceProvider,
+            IThreadingContext threadingContext)
+        {
+            _serviceProvider = serviceProvider;
+            _threadingContext = threadingContext;
+        }
 
-        public override DefinitionItem? GetThirdPartyDefinitionItem(
+        public override async Task<DefinitionItem?> GetThirdPartyDefinitionItemAsync(
             Solution solution, DefinitionItem definitionItem, CancellationToken cancellationToken)
         {
             var symbolNavigationService = solution.Workspace.Services.GetRequiredService<ISymbolNavigationService>();
-            if (!symbolNavigationService.WouldNavigateToSymbol(
-                    definitionItem, solution, cancellationToken,
-                    out var filePath, out var lineNumber, out var charOffset))
-            {
+            var result = await symbolNavigationService.WouldNavigateToSymbolAsync(definitionItem, solution, cancellationToken).ConfigureAwait(false);
+            if (result is not var (filePath, lineNumber, charOffset))
                 return null;
-            }
 
             var displayParts = GetDisplayParts(filePath, lineNumber, charOffset);
```

Comment:
❗ Need to switch to the main thread before this line

Reply:
terrifying.

Reply:
➡️ Fixed in 7517e32a050d65711ad3efca0274c2d3d4a553b4